.github/workflows/build-test.yaml

```diff
@@ -80,74 +160,168 @@ jobs:
       - name: Setup Node.js environment
         uses: actions/setup-node@v3
         with:
-          node-version: '17.x'
+          node-version: '18.x'
+          cache: yarn
+          cache-dependency-path: web/yarn.lock
 
+      - name: Cache node_modules
+        uses: actions/cache@v3
+        with:
+          path: ${{ github.workspace }}/web/node_modules
+          key: ${{ runner.os }}-${{ runner.arch }}-yarn-node_modules-${{ hashFiles('web/yarn.lock') }}
+          
       - name: Build web
         env:
           GIT_TAG: ${{ needs.generate-tag.outputs.tag }}
-        run: mapfile -t envs < <(grep -v '#.*' < .image.env) && export "${envs[@]}" && make -C web deps lint test-unit build-kotsadm
+        run: mapfile -t envs < <(grep -v '#.*' < .image.env) && export "${envs[@]}" && make -C web build-kotsadm
 
       - name: Upload web artifact
         uses: actions/upload-artifact@v3
         with:
           name: web
           path: ./web/dist
 
+  deps-kots:
+    runs-on: ubuntu-20.04
+    needs: [ can-run-ci ]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
 
-  build-kots:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '^1.19.2'
+          cache: true
+
+      - run: go mod download
+
+  vet-kots:
+    runs-on: ubuntu-20.04
+    needs: [ can-run-ci, deps-kots ]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '^1.19.2'
+          cache: true
+ 
+      - name: vet
+        run: make vet
+  
+  ci-test-kots:
```

Comment:
this step should probably be added to the final `validate-success` step so that it blocks on failure